### PR TITLE
Channel variant

### PR DESCRIPTION
**Pull request by philwhiles**
CCSvc now has a configurable 'channel' which can be either CC or AD.
The channel is used to determine the single authenticated users role, and the WebSecurityConfig now uses the role to allow or forbid access to all endpoints explicitly.
The channel is also used when sending to the event publisher, such that events sent from the service in CC channel mode will be identified as coming from CC, and AD ... AD

### CODE REVIEW

**Comment by rpcatling** [comment]
Does this need or have a JIRA associated with it ?

**Comment by philwhiles** [comment]
I had no JIRA access when I started it and used CR-806 in commits later as that was the ticket that kicked this off

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/CaseEndpoint.java [hunk 1]
```diff
@@ -249,4 +256,116 @@ public void initBinder(WebDataBinder binder) {
...
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<ResponseDTO> modifyCase(
+      @PathVariable(value = "caseId") final UUID caseId,
+      @Valid @RequestBody ModifyCaseRequestDTO requestBodyDTO)
+      throws CTPException {
+
+    log.with("requestBody", requestBodyDTO).info("Entering PUT modifyCase");
+
+    if (!caseId.equals(requestBodyDTO.getCaseId())) {
+      log.with(caseId)
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/CaseEndpoint.java:283]: Will leave that until actual implementation time. Can only do so much to test what are currently dummy endpoints, until the real code comes along

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/WebSecurityConfig.java [hunk 2]
```diff
@@ -17,29 +17,49 @@
...
+        .hasRole("AD")
+        .antMatchers("/cases")
+        .hasRole("CC")
+        .antMatchers("/cases/uprn/**")
+        .access("hasRole('CC') or hasRole('AD')")
+        .antMatchers("/addresses")
+        .access("hasRole('CC') or hasRole('AD')")
+        .antMatchers("/addresses/postcode")
+        .access("hasRole('CC') or hasRole('AD')")
+        .antMatchers("/cases/**")
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/WebSecurityConfig.java:41]: Will remove ineffectual /case/this/that match lines after the /case/**

Re: formatting - if only the maven formatter honoured the no format comments...

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/EndpointSecurityTest.java [hunk 3]
```diff
@@ -6,16 +6,34 @@
...
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseType;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.EstabType;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.ModifyCaseRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.NewCaseRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.PostalFulfilmentRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.Reason;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.RefusalRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.Region;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.ResponseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.SMSFulfilmentRequestDTO;
```
[the remark this answers is not in the document]
Comment (philwhiles) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/EndpointSecurityTest.java:37]: Have made the EndpointSecurityTest abstract as no point it running on its own

**Pull request merged**